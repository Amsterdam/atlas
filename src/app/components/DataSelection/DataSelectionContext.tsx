import { useMemo, useState } from 'react'
import type { Dispatch, FunctionComponent, SetStateAction } from 'react'
import type { DataSelectionFilters } from '../../pages/MapPage/query-params'
import { dataSelectionFiltersParam, polygonParam } from '../../pages/MapPage/query-params'
import createNamedContext from '../../utils/createNamedContext'
import useParam from '../../utils/useParam'
import useRequiredContext from '../../utils/useRequiredContext'
import type { ActiveFilter, AvailableFilter } from './types'
import useLegacyDataselectionConfig from './useLegacyDataselectionConfig'

type Action<T extends keyof DataSelectionContextProps> = Dispatch<
  SetStateAction<DataSelectionContextProps[T]>
>

export interface DataSelectionContextProps {
  availableFilters: AvailableFilter[]
  addFilter: (filter: DataSelectionFilters) => void
  removeFilter: (key: string) => void
  totalResults: number
  activeFilters: ActiveFilter[]
  setTotalResults: Action<'totalResults'>
  setAvailableFilters: Action<'availableFilters'>
  distanceText?: string
  drawToolLocked: boolean
  setDrawToolLocked: Action<'drawToolLocked'>
  setDistanceText: Action<'distanceText'>
}

const DataSelectionContext = createNamedContext<DataSelectionContextProps | null>(
  'DataSelection',
  null,
)

const DataSelectionProvider: FunctionComponent = ({ children }) => {
  const [activeParamFilters, setActiveParamFilters] = useParam(dataSelectionFiltersParam)
  const [polygon] = useParam(polygonParam)
  const [distanceText, setDistanceText] = useState<string>()
  const [drawToolLocked, setDrawToolLocked] = useState(false)
  const [availableFilters, setAvailableFilters] = useState<
    DataSelectionContextProps['availableFilters']
  >([])
  const [totalResults, setTotalResults] = useState(0)
  const { currentDatasetConfig } = useLegacyDataselectionConfig()

  const addFilter = (filter: DataSelectionFilters) => {
    setActiveParamFilters({
      ...activeParamFilters,
      ...filter,
    })
  }
  const removeFilter = (key: string) => {
    // @ts-ignore
    const { [key]: omit, ...rest } = activeParamFilters
    setActiveParamFilters(Object.keys(rest).length ? rest : null)
  }

  const shape = useMemo(
    () =>
      polygon?.polygon?.length
        ? JSON.stringify(polygon.polygon.map(({ lat, lng }) => [lng, lat]))
        : null,
    [polygon],
  )

  const activeFilters = useMemo(() => {
    const filters =
      activeParamFilters && currentDatasetConfig
        ? Object.entries(activeParamFilters).reduce<ActiveFilter[]>((acc, [key, val]) => {
            const label = currentDatasetConfig.FILTERS.find(({ slug }) => slug === key)?.label
            if (label) {
              let value = val
              if (key === 'sbi_code') {
                value = val.replace(/['[\]]/g, '')
              }
              acc.push({ key, value, label: `${label}: ${value}` })
            }
            return acc
          }, [])
        : []

    if (shape) {
      filters.push({
        key: 'shape',
        value: shape,
        label: 'Ingetekende vorm op kaart',
      })
    }

    return filters
  }, [currentDatasetConfig, activeParamFilters, shape])

  return (
    <DataSelectionContext.Provider
      value={{
        // Only provide the token if the user does not have an authenticated session.
        availableFilters,
        activeFilters,
        addFilter,
        removeFilter,
        totalResults,
        setTotalResults,
        setAvailableFilters,
        distanceText,
        setDistanceText,
        drawToolLocked,
        setDrawToolLocked,
      }}
    >
      {children}
    </DataSelectionContext.Provider>
  )
}

export { DataSelectionProvider }

export function useDataSelection() {
  return useRequiredContext(DataSelectionContext)
}
