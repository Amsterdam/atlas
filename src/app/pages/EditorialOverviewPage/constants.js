import PAGES from '../../pages'
import {
  toArticleDetail,
  toPublicationDetail,
  toSpecialDetail,
} from '../../../store/redux-first-router/actions'
import { TYPES } from '../../../shared/config/cms.config'
import {
  articleSearchQuery,
  publicationSearchQuery,
  specialSearchQuery,
} from '../SearchPage/graphql.config'

export const EDITORIAL_TYPES = {
  [PAGES.ARTICLES]: TYPES.ARTICLE,
  [PAGES.ARTICLE_SEARCH]: TYPES.ARTICLE,
  [PAGES.PUBLICATIONS]: TYPES.PUBLICATION,
  [PAGES.PUBLICATION_SEARCH]: TYPES.PUBLICATION,
  [PAGES.SPECIALS]: TYPES.SPECIAL,
  [PAGES.SPECIAL_SEARCH]: TYPES.SPECIAL,
}

export const EDITORIAL_TITLES = {
  [TYPES.ARTICLE]: 'Artikelen',
  [TYPES.PUBLICATION]: 'Publicaties',
  [TYPES.SPECIAL]: 'In beeld',
}

export const EDITORIAL_DETAIL_ACTIONS = {
  [TYPES.ARTICLE]: toArticleDetail,
  [TYPES.PUBLICATION]: toPublicationDetail,
  [TYPES.SPECIAL]: toSpecialDetail,
}

export const GRAPHQL_CONFIG = {
  [PAGES.ARTICLES]: {
    resolver: 'articleSearch',
    query: articleSearchQuery,
  },
  [PAGES.PUBLICATIONS]: {
    resolver: 'publicationSearch',
    query: publicationSearchQuery,
  },
  [PAGES.SPECIALS]: {
    resolver: 'specialSearch',
    query: specialSearchQuery,
  },
}

// Logic is that we don't show metadata in an editorial detail page
export const EDITORIAL_FIELD_TYPE_VALUES = {
  CONTENT: 'content',
}
