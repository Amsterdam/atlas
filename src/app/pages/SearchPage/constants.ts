export enum SearchType {
  Search = 'search',
  Data = 'data',
  Dataset = 'dataset',
  Map = 'map',
}

export default SearchType
