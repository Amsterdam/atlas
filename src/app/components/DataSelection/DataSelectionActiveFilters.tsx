import { useDispatch } from 'react-redux'
import { FilterTag } from '@amsterdam/asc-ui'
import styled from 'styled-components'
import { removeGeometryFilter } from '../../../shared/ducks/data-selection/actions'
import { useDataSelection } from './DataSelectionContext'
import useParam from '../../utils/useParam'
import { polygonParam } from '../../pages/MapPage/query-params'

const List = styled.ul`
  display: flex;
  flex-wrap: wrap;
  list-style: none;
`

const DataSelectionActiveFilters = () => {
  const dispatch = useDispatch()
  const { removeFilter, activeFilters } = useDataSelection()
  const [, setPolygon] = useParam(polygonParam)
  return activeFilters.length ? (
    <List data-testid="activeFilters">
      {activeFilters.map(({ key, label, value }) => (
        <li key={key}>
          <FilterTag
            type="button"
            aria-label={`Filter verwijderen: ${label}: ${value}`}
            onClick={() => {
              if (key === 'shape') {
                setPolygon(null)
                dispatch(removeGeometryFilter())
              } else {
                removeFilter(key)
              }
            }}
            data-testid="activeFiltersItem"
          >
            {label}
          </FilterTag>
        </li>
      ))}
    </List>
  ) : null
}

export default DataSelectionActiveFilters
