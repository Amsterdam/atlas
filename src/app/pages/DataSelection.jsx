import React from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import MapContainer from '../../map/containers/map/MapContainer';
import NewDataSelection from '../components/DataSelection/DataSelection';
import { getDataSelectionView } from '../../shared/ducks/data-selection/selectors';
import SplitScreen from '../components/SplitScreen/SplitScreen';
import { VIEWS } from '../../shared/ducks/data-selection/constants';
import { setView } from '../../shared/ducks/data-selection/actions';


/* istanbul ignore next */ // TODO: refactor, test
const DataSelection = ({ view, toggleList, toggleMap }) => {
  if (view === VIEWS.MAP) {
    return (<MapContainer isFullscreen toggleFullscreen={toggleList} />);
  }
  if (view == VIEWS.LIST) {
    return (
      <SplitScreen
        leftComponent={(
          <MapContainer isFullscreen={false} toggleFullscreen={toggleMap} />
        )}
        rightComponent={(
          <NewDataSelection />
        )}
      />
    );
  }

  return (
    <NewDataSelection />
  );
};

DataSelection.propTypes = {
  view: PropTypes.string.isRequired,
  toggleList: PropTypes.func.isRequired,
  toggleMap: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  view: getDataSelectionView(state)
});

const mapDispatchToProps = (dispatch) => bindActionCreators({
  toggleList: () => setView(VIEWS.LIST),
  toggleMap: () => setView(VIEWS.MAP)
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(DataSelection);
