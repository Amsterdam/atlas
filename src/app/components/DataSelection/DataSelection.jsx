import React from 'react';
import PropTypes from 'prop-types';

import { connect } from 'react-redux';
import classNames from 'classnames';
import { bindActionCreators } from 'redux';
import {
getDataSelection,
getDataSelectionResult,
setPage as setDatasetPage,
VIEWS
} from '../../../shared/ducks/data-selection/data-selection';
import DATA_SELECTION_CONFIG from '../../../shared/services/data-selection/data-selection-config';
import { getUser } from '../../../shared/ducks/user/user';
import NotAuthorizedPanel from '../PanelMessages/NotAuthorizedMessage';
import LoadingIndicator from '../../../shared/components/loading-indicator/LoadingIndicator';
import { getActiveFilters } from '../../../shared/ducks/filters/filters';
import DataSelectionActiveFilters from '../../containers/DataSelectionActiveFiltersContainer';
import MaxPageMessage from '../PanelMessages/MaxPageMessage';
import AngularWrapper from '../../../react-angular/AngularWrapper';

const DataSelection = ({
  view,
  isLoading,
  dataset,
  activeFilters,
  user,
  setPage,
  authError,
  results: {
    numberOfRecords,
    filters: availableFilters,
    numberOfPages,
    data
  },
  page: currentPage
}) => {
  if (isLoading || (!numberOfRecords && !authError)) {
    return <LoadingIndicator />;
  }

  // Local state
  const showHeader = (view === VIEWS.LIST || !isLoading);
  const showFilters = (view !== VIEWS.LIST) && numberOfRecords > 0;
  const { MAX_AVAILABLE_PAGES, MAX_NUMBER_OF_CLUSTERED_MARKERS } =
    DATA_SELECTION_CONFIG.datasets[dataset];

  const showMessageMaxPages = MAX_AVAILABLE_PAGES && currentPage > MAX_AVAILABLE_PAGES;
  const showMessageClusteredMarkers =
    (view === VIEWS.LIST) && numberOfRecords > MAX_NUMBER_OF_CLUSTERED_MARKERS;

  const widthClass = classNames({
    'u-col-sm--12': !showFilters,
    'u-col-sm--9': showFilters
  });

  return (
    <div className="c-data-selection">
      <div className="qa-data-selection-content">
        <AngularWrapper
          moduleName={'dpDataSelectionHeaderWrapper'}
          component="dpDataSelectionHeader"
          dependencies={['atlas']}
          bindings={{
            dataset,
            availableFilters,
            filters: activeFilters,
            numberOfRecords,
            showHeader,
            user,
            view
          }}
        />

        <DataSelectionActiveFilters />

        {(!authError) && (
          <div className="u-grid qa-data-grid">
            <div className="u-row">
              {showFilters && (
                <div className="u-col-sm--3 c-data-selection__available-filters">
                  {(dataset === 'hr') && (
                    <AngularWrapper
                      moduleName="dpSbiFilterWrapper"
                      component="dpSbiFilter"
                      dependencies={['atlas']}
                      bindings={{
                        availableFilters,
                        filters: activeFilters
                      }}
                    />
                  )}

                  <AngularWrapper
                    moduleName="dpDataSelectionAvailableFiltersWrapper"
                    component="dpDataSelectionAvailableFilters"
                    dependencies={['atlas']}
                    bindings={{
                      availableFilters,
                      activeFilters
                    }}
                    interpolateBindings={{
                      dataset
                    }}
                  />
                </div>
              )}
              <div className={widthClass}>
                {showMessageMaxPages && (
                  <MaxPageMessage maxAvailablePages={MAX_AVAILABLE_PAGES} />
                )}
                {showMessageClusteredMarkers && (
                  <AngularWrapper
                    moduleName={'dpPanelWrapper'}
                    component="dpPanel"
                    dependencies={['atlas']}
                    bindings={{
                      isPanelVisible: true,
                      canClose: true
                    }}
                    interpolateBindings={{
                      type: 'warning'
                    }}
                  >
                    <div className="qa-message-clustered-markers">
                      <p className="c-panel__paragraph">Deze resultaten worden niet getoond op
                        de
                        kaart, omdat deze niet meer
                        dan {MAX_NUMBER_OF_CLUSTERED_MARKERS} resultaten tegelijk kan
                        weergeven (om technische redenen).</p>
                      <p className="c-panel__paragraph">Tip: Bekijk de lijst resultaten in
                        kleinere delen. Dit kan door een voor een filtercriteria toe te voegen
                        (bijv. de verschillende wijken uit de selectie).</p>
                    </div>
                  </AngularWrapper>
                )}


                <div>
                  {view === VIEWS.TABLE && (
                    <AngularWrapper
                      moduleName={'dpDataSelectionTableWrapper'}
                      component="dpDataSelectionTable"
                      dependencies={['atlas']}
                      bindings={{
                        content: data,
                        dataset
                      }}
                    />
                  )}
                  {view === VIEWS.LIST && (
                    <AngularWrapper
                      moduleName={'dpDataSelectionListWrapper'}
                      component="dpDataSelectionList"
                      dependencies={['atlas']}
                      bindings={{
                        content: data
                      }}
                    />
                  )}
                </div>
                <AngularWrapper
                  moduleName={'dpDataSelectionPaginationWrapper'}
                  component="dpDataSelectionPagination"
                  dependencies={['atlas']}
                  bindings={{
                    currentPage,
                    numberOfPages,
                    setPage
                  }}
                />
              </div>
            </div>
          </div>
        )}

        {authError && (
          <NotAuthorizedPanel />
        )}
      </div>
    </div>
  );
};

DataSelection.defaultProps = {
  results: {}
};

DataSelection.propTypes = {
  view: PropTypes.string.isRequired,
  isLoading: PropTypes.bool.isRequired,
  dataset: PropTypes.string.isRequired,
  activeFilters: PropTypes.shape({}).isRequired,
  user: PropTypes.shape({}).isRequired,
  authError: PropTypes.bool.isRequired,
  page: PropTypes.number.isRequired,
  setPage: PropTypes.func.isRequired,
  results: PropTypes.shape({
    numberOfRecords: PropTypes.number,
    filters: PropTypes.arrayOf(PropTypes.object),
    numberOfPages: PropTypes.number,
    data: PropTypes.object
  })
};

const mapStateToProps = (state) => {
  const { view, isLoading, dataset, authError, page } = getDataSelection(state);
  return ({
    view,
    isLoading,
    dataset,
    authError,
    page,
    activeFilters: getActiveFilters(state),
    results: getDataSelectionResult(state),
    user: getUser(state)
  });
};

const mapDispatchToProps = (dispatch) => bindActionCreators({
  setPage: setDatasetPage
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(DataSelection);
