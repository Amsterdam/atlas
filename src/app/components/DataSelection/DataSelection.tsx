import { Alert, Heading, Paragraph, themeColor, themeSpacing } from '@amsterdam/asc-ui'
import { useSelector } from 'react-redux'
import usePromise, { isPending, isRejected } from '@amsterdam/use-promise'
import styled, { css } from 'styled-components'
import { FunctionComponent } from 'react'
import { getDataSelectionPage } from '../../../shared/ducks/data-selection/selectors'
import DataSelectionActiveFilters from './DataSelectionActiveFilters'
import { ViewMode, viewParam } from '../../pages/MapPage/query-params'
import useParam from '../../utils/useParam'
import LoginLink from '../Links/LoginLink/LoginLink'
import LoadingSpinner from '../LoadingSpinner/LoadingSpinner'
import ShareBar from '../ShareBar/ShareBar'
import DataSelectionList from './DataSelectionList'
import DataSelectionTable from './DataSelectionTable'
import LegacyPagination from './LegacyPagination'
import DataSelectionFilters from './DataSelectionFilters'
import DataSelectionSbiFilters from './DataSelectionSbiFilters'
import useFetchLegacyDataSelectionData from './useFetchLegacyDataSelectionData'
import { useDataSelection } from './DataSelectionContext'
import DataSelectionHeader from './DataSelectionHeader'
import { AuthError } from '../../../shared/services/api/customError'
import useFetchLegacyDataSelectionMarkers from './useFetchLegacyDataSelectionMarkers'

const StyledAlert = styled(Alert)`
  margin-bottom: ${themeSpacing(5)};
`

const StyledContainer = styled.section<{ view: ViewMode }>`
  display: flex;
  flex-direction: column;
  margin: ${themeSpacing(4, 0)};
  padding: ${themeSpacing(0, 4)};
  background-color: ${themeColor('tint', 'level1')};

  ${({ view }) =>
    view === ViewMode.Full &&
    css`
      overflow-x: auto;
    `}
`

const Wrapper = styled.div<{ view: ViewMode }>`
  margin-bottom: ${themeSpacing(4)};
  ${({ view }) =>
    view === ViewMode.Full &&
    css`
      display: grid;
      grid-template-columns: 25% 75%;
      grid-template-rows: 2fr;
      grid-column-gap: ${themeSpacing(2)};
      grid-row-gap: 0;
    `}
`

const Footer = styled.footer`
  grid-column: 2;
`

interface DataSelectionContentProps {
  view: ViewMode
}

const DataSelectionContent: FunctionComponent<DataSelectionContentProps> = ({ view }) => {
  const page = useSelector(getDataSelectionPage)
  const { activeFilters, totalResults } = useDataSelection()
  const { fetchData } = useFetchLegacyDataSelectionData()
  const { fetchMarkers } = useFetchLegacyDataSelectionMarkers()

  const result = usePromise(
    (signal) => fetchData(signal),
    [
      view, // view can change: table or map view AKA Full or Split mode
      page, // Pagination page
      activeFilters, // Enabling or disabling filters
    ],
  )

  usePromise((signal) => fetchMarkers(signal), [result])

  if (isPending(result)) {
    return <LoadingSpinner />
  }

  if (isRejected(result)) {
    if (result.reason instanceof AuthError) {
      return (
        <Alert level="info" dismissible>
          <Paragraph>{result.reason.message}</Paragraph>
          <LoginLink />
        </Alert>
      )
    }

    return (
      <Alert level="warning" dismissible>
        {result.reason.message}
      </Alert>
    )
  }

  const { numberOfPages, data, showSbiFilers, messageMaxPages, messageClusteredMarkers } =
    result.value

  const showFilters = view === ViewMode.Full && totalResults > 0
  return (
    <>
      <DataSelectionActiveFilters />

      {!totalResults && (
        <>
          <Paragraph>Geen resultaten van deze soort</Paragraph>
          <Paragraph>Tip: verwijder een of meer criteria</Paragraph>
        </>
      )}

      <Wrapper view={view}>
        {showFilters && (
          <div>
            {showSbiFilers && <DataSelectionSbiFilters />}
            <DataSelectionFilters />
          </div>
        )}
        <div>
          {messageMaxPages && (
            <StyledAlert level="info" dismissible>
              <Heading forwardedAs="h3">Deze pagina kan niet worden getoond</Heading>
              <Paragraph>{messageMaxPages}</Paragraph>
              <Paragraph>
                Tip: Gebruik de download-knop om alle resultaten te bekijken. Of voeg meer
                filtercriteria toe voor specifiekere resultaten.
              </Paragraph>
            </StyledAlert>
          )}
          {messageClusteredMarkers && (
            <StyledAlert level="info">
              <Paragraph>{messageClusteredMarkers}</Paragraph>
              <Paragraph>
                Tip: Bekijk de lijst resultaten in kleinere delen. Dit kan door een voor een
                filtercriteria toe te voegen (bijv. de verschillende wijken uit de selectie).
              </Paragraph>
            </StyledAlert>
          )}

          {totalResults > 0 ? (
            <div>
              {view === ViewMode.Full && <DataSelectionTable content={data} />}
              {view === ViewMode.Split && <DataSelectionList content={data} />}
            </div>
          ) : null}
        </div>
        <Footer>
          <LegacyPagination currentPage={page} numberOfPages={numberOfPages} />
          {view === ViewMode.Full && <ShareBar />}
        </Footer>
      </Wrapper>
    </>
  )
}

const DataSelection = () => {
  const [view] = useParam(viewParam)
  return (
    <StyledContainer view={view} className="c-data-selection">
      <div className="c-data-selection-content">
        <DataSelectionHeader />
        <DataSelectionContent view={view} />
      </div>
    </StyledContainer>
  )
}

export default DataSelection
