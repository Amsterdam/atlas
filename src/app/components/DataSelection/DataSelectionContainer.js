import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { setPage as setDatasetPage } from '../../../shared/ducks/data-selection/actions';
import { getUser, getUserScopes } from '../../../shared/ducks/user/user';
import { getFilters } from '../../../shared/ducks/filters/filters';
import {
  getDataSelection,
  getDataSelectionResult,
  getGeometryFilter
} from '../../../shared/ducks/data-selection/selectors';
import { getViewMode } from '../../../shared/ducks/ui/ui';
import DataSelection from './DataSelection';

const mapStateToProps = (state) => {
  const { isLoading, dataset, authError, page } = getDataSelection(state);
  return ({
    isLoading,
    dataset,
    authError,
    page,
    view: getViewMode(state),
    geometryFilter: getGeometryFilter(state),
    activeFilters: getFilters(state),
    results: getDataSelectionResult(state),
    user: getUser(state),
    userScopes: getUserScopes(state)
  });
};

const mapDispatchToProps = (dispatch) => bindActionCreators({
  setPage: setDatasetPage
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(DataSelection);
