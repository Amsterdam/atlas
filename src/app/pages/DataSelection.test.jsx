import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import DataSelection from './DataSelection';
import { getDataSelectionView } from '../../shared/ducks/data-selection/selectors';
import { setView } from '../../shared/ducks/data-selection/actions';
import { VIEWS } from '../../shared/ducks/data-selection/constants';

jest.mock('../../shared/ducks/data-selection/selectors');
jest.mock('../../shared/ducks/data-selection/actions');

const initialState = {
};

describe('DataSelection', () => {
  it(`should render as ${VIEWS.LIST}`, () => {
    getDataSelectionView.mockReturnValue(VIEWS.LIST);
    setView.mockReturnValue(VIEWS.LIST);

    const store = configureMockStore()({ ...initialState });
    const component = shallow(<DataSelection view />, { context: { store } }).dive();
    expect(component).toMatchSnapshot();
  });

  it(`should render as ${VIEWS.MAP}`, () => {
    getDataSelectionView.mockReturnValue(VIEWS.MAP);
    setView.mockReturnValue(VIEWS.MAP);

    const store = configureMockStore()({ ...initialState });
    const component = shallow(<DataSelection view />, { context: { store } }).dive();
    expect(component).toMatchSnapshot();
  });
});
