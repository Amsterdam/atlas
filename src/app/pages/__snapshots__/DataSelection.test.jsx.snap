// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSelection should render as LIST 1`] = `
<SplitScreen
  leftComponent={
    <Connect(MapContainer)
      isFullscreen={false}
      toggleFullscreen={[Function]}
    />
  }
  rightComponent={<Connect(DataSelection) />}
/>
`;

exports[`DataSelection should render as MAP 1`] = `
<Connect(MapContainer)
  isFullscreen={true}
  toggleFullscreen={[Function]}
/>
`;
